### PR TITLE
Bump terraform collection to fix race with waiting for ssh

### PR DESCRIPTION
**Pull request by sjpb**
See https://github.com/azimuth-cloud/ansible-collection-terraform/pull/16

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ collections:
     version: 2.1.0
   - name: https://github.com/stackhpc/ansible-collection-terraform
```
Comment (mkjpryor) [inline requirements.yml:46]: This needs to change to `azimuth-cloud`, as do any uses of the roles from the collection.

Comment (mkjpryor) [reply requirements.yml:46]: I'm actually surprised it works as it is? Probably the CI doesn't test the CaaS Terraform?

**Review by mkjpryor — CHANGES_REQUESTED** [review]

File: requirements.yml (hunk 1 above)
Comment (sjpb) [reply requirements.yml]: Yeah CI doesn't test this, I just ran into this while testing manually in a dev azimuth :flushed:

**Comment by sjpb** [comment]
Tested in azimuth using:
```
azimuth_caas_stackhpc_slurm_appliance_git_version: update/tf-collection
azimuth_caas_stackhpc_slurm_appliance_image: 5e9af448-ab2b-4d48-bb63-9ab103a85e80 #openhpc-RL9-241009-1523-354b048a # current main @ c7f3b24
```

worked fine.

File: ansible/roles/cluster_infra/tasks/main.yml [hunk 2]
```diff
@@ -10,20 +10,20 @@
...
+      register: cluster_floating_ip_info
 
     - name: Set floating IP address fact
       set_fact:
-        cluster_floating_ip_address: "{{ os_floating_ip_info.floating_ip_address }}"
-  when: cluster_floating_ip is defined
+        cluster_floating_ip_address: "{{ cluster_floating_ip_info.floating_ip.floating_ip_address }}"
+      when:
+        - cluster_floating_ip is defined
+        - cluster_floating_ip
```
Comment (mkjpryor) [inline ansible/roles/cluster_infra/tasks/main.yml:22]: This `when` should be on the `block` I think

**Review by mkjpryor — CHANGES_REQUESTED** [review]

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**